build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -186,7 +189,9 @@ class WatchImpl implements BuildState {
   final Set<AssetId> _expectedDeletes = <AssetId>{};
 
   FinalizedReader? _reader;
-  FinalizedReader get reader => _reader!;
+
+  /// Only non-null after [ready] completes without error.
+  FinalizedReader? get reader => _reader;
```

Comment:
What do you think of combining this with `ready`?

`Future<FinalizedReader> get ready`

(or keep the name `reader` but make it a `Future` and drop the `ready` field)

Reply:
Hmm ya it might make more sense to just make it a Future, idk why it wasn't one already.